src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenArray = withChildren
+    ? people.filter(person => person.sex === 'f'
+      && people.some(child => child.mother === person.name))
+    : people.filter(person => person.sex === 'f');
+
+  const womenAges = womenArray.map(person => person.died - person.born);
+  const sumOfMenAges = womenAges.reduce((sum, age) => sum + age, 0);
```

Comment:
Are you sure that it's sumOf**Men**Ages?
By the way, you have the same logic for men and women. It would be great to create a function helper for calculating an average age of the array